### PR TITLE
Feature/55 review dynamic data

### PR DESCRIPTION
**Pull request by softwarejimenez**

### CODE REVIEW

File: srcDoc/release_notes.rst [hunk 1]
```diff
@@ -62,6 +62,22 @@ following command-line parameters:
     --cpp-build
     --cs-build
 
+What's Fixed in Master
+~~~~~~~~~~~~~~~~~~~~~~
+
+Review Dynamic Data Send() and Receive() operations (#55)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+*RTI Perftest* has optimize the `Send()` and `Received()` methods by:
```
Comment (javmorcas) [inline srcDoc/release_notes.rst]: has optimized*

but I would say:

The `Send()` and `Receive()` functions for *Dynamic data* have been optimized by:

File: srcDoc/release_notes.rst [hunk 2]
```diff
@@ -62,6 +62,22 @@ following command-line parameters:
...
+
+- Specifying the member_id in the `get_T()` and `set_T()` methods instead of
+  `DDS_DYNAMIC_DATA_MEMBER_ID_UNSPECIFIED`.
+- Creating a `std::map<std::string, int>` for the DDS_DynamicDataMemberIds.
+- Set the content of `bind_data`, using `set_octet_array` when the message_size
+  changes.
+- Clearing the contents of DDS_DynamicData object, calling `clear_all_members()`
+  when the size change.
```
Comment (javmorcas) [inline srcDoc/release_notes.rst]: changes

File: srcDoc/release_notes.rst [hunk 3]
```diff
@@ -62,6 +62,22 @@ following command-line parameters:
...
+~~~~~~~~~~~~~~~~~~~~~~
+
+Review Dynamic Data Send() and Receive() operations (#55)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+*RTI Perftest* has optimize the `Send()` and `Received()` methods by:
+
+- Specifying the member_id in the `get_T()` and `set_T()` methods instead of
+  `DDS_DYNAMIC_DATA_MEMBER_ID_UNSPECIFIED`.
+- Creating a `std::map<std::string, int>` for the DDS_DynamicDataMemberIds.
```
Comment (javmorcas) [inline srcDoc/release_notes.rst]: Maybe you don't need to include this, seems like a very specific detail for the implementation.

File: srcDoc/release_notes.rst [hunk 4]
```diff
@@ -62,6 +62,22 @@ following command-line parameters:
...
+- Set the content of `bind_data`, using `set_octet_array` when the message_size
+  changes.
+- Clearing the contents of DDS_DynamicData object, calling `clear_all_members()`
+  when the size change.
+
```
Comment (javmorcas) [inline srcDoc/release_notes.rst]: ...

The result of these optimizations is that *RTI Perftest* now minimizes the time employed in the application-related tasks, therefore maximizing the time for sending and receiving calls.

File: srcCpp/RTIDDSImpl.cxx [hunk 5]
```diff
@@ -1020,61 +1028,61 @@ class RTIDynamicDataPublisher : public IMessagingWriter
...
+        DDS_Octet key_octets[KEY_SIZE];
         long key = 0;
 
-        data.clear_all_members();
-
+        if (_last_message_size != message.size) {
+            //Cannot use data.clear_member("bind_data") because:
+            //DDS_DynamicData_clear_member:unsupported for non-sparse types
+            data.clear_all_members();
+            retcode = data.set_octet_array(
```
Comment (javmorcas) [inline srcCpp/RTIDDSImpl.cxx]: Check compatibility non-dynamic <-> dynamic

File: srcCpp/RTIDDSImpl.h [hunk 6]
```diff
@@ -15,6 +15,26 @@
 
 #define RTIPERFTEST_MAX_PEERS 1024
 
+struct dynamicDataMembersId{
```
[the remark this answers is not in the document]
Comment (javmorcas) [reply srcCpp/RTIDDSImpl.h]: Use a class

File: srcCpp/RTIDDSImpl.cxx (hunk 5 above)
Comment (javmorcas) [reply srcCpp/RTIDDSImpl.cxx]: What is the conclusion on this? @softwarejimenez?

Comment (softwarejimenez) [reply srcCpp/RTIDDSImpl.cxx]: I have run a couple of examples about compatibility of non-dynamic <-> dynamic and it works fine using set_octect_array

**Comment by softwarejimenez** [comment]
@javmorcas @alfonso-soto Hi team. I have finished the full changes. Can you review them? 
Thank you in advance.

File: srcCpp/RTIDDSImpl.cxx (hunk 5 above)
Comment (softwarejimenez) [reply srcCpp/RTIDDSImpl.cxx]: use set_seq

Comment (softwarejimenez) [reply srcCpp/RTIDDSImpl.cxx]: Done

File: srcDoc/release_notes.rst [hunk 7]
```diff
@@ -62,6 +62,26 @@ following command-line parameters:
     --cpp-build
     --cs-build
 
+What's Fixed in Master
+~~~~~~~~~~~~~~~~~~~~~~
+
+Review Dynamic Data Send() and Receive() operations (#55)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The `Send()` and `Received()` for Dynamic Data have been optimized by:
```
Comment (javmorcas) [inline srcDoc/release_notes.rst]: The *Dynamic Data* `Send()` and `Received()` functions have been optimized by:

File: srcDoc/release_notes.rst [hunk 8]
```diff
@@ -62,6 +62,26 @@ following command-line parameters:
     --cpp-build
     --cs-build
 
+What's Fixed in Master
+~~~~~~~~~~~~~~~~~~~~~~
+
+Review Dynamic Data Send() and Receive() operations (#55)
```
Comment (javmorcas) [inline srcDoc/release_notes.rst]: Improve *Dynamic Data* Send() and Receive() operations (#55)

File: srcDoc/release_notes.rst (hunk 7 above)
Comment (javmorcas) [reply srcDoc/release_notes.rst]: Less technical, but i think it covers the main idea of making this comparable.

File: srcDoc/release_notes.rst (hunk 8 above)
Comment (softwarejimenez) [reply srcDoc/release_notes.rst]: Done

**Pull request merged**